api/src/org/labkey/api/view/RedirectorServlet.java

```diff
@@ -27,11 +28,20 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
         if ("get".equalsIgnoreCase(request.getMethod()))
         {
             // Send a redirect to let the client know there's a new preferred URL
-            String originalUrl = request.getRequestURL().toString();
-            String redirectUrl = originalUrl.replaceFirst(_legacyContextPath, "");
+            StringBuffer rawUrl = request.getRequestURL();
+            // getRequestURL() doesn't include GET parameters, so append them if needed
+            if (request.getQueryString() != null)
+            {
+                rawUrl.append("?").append(request.getQueryString());
+            }
+            URL originalUrl = new URL(rawUrl.toString());
+            URL redirectUrl = new URL(originalUrl.getProtocol(),
+                    originalUrl.getHost(),
+                    originalUrl.getPort(),
+                    originalUrl.getFile().replaceFirst(_legacyContextPath, ""));
```

Comment:
Should we ensure that we only replace the legacy context path if the URL file begins with it?
```suggestion
                    originalUrl.getFile().replaceFirst("^" + _legacyContextPath, ""));
```

Which leads me to the thought: What if there's a project or folder that happens to match the legacy context path? The legacy URL `/labkey/labkey/project-begin.view` would redirect to what you want but the "correct" URL would be broken.

Reply:
This is indeed a limitation to the current approach but more complicated to try to redirect only if we're otherwise going to serve a 404. If someone hits it, they can keep deploying to their old context path and we can regroup on the best approach to migrate them.

Reply:
This servlet is registered for URLs that start with the legacy context path, so we don't need to separately check. Your proposed change would be fine, but it won't have a functional difference.

https://github.com/LabKey/platform/blob/5649b28c6498d0ab292bdadd95218cc966940e32/api/src/org/labkey/api/ApiModule.java#L261